core/trie/trie.go

```diff
@@ -154,7 +166,17 @@ func (t *Trie) Get(key *felt.Felt) (*felt.Felt, error) {
 
 // Put updates the corresponding `value` for a `key`
 func (t *Trie) Put(key *felt.Felt, value *felt.Felt) error {
-	// Todo: check key is not bigger than max key value for a trie height.
+	var x big.Int
+	x.SetInt64(1).Lsh(&x, t.height)
```

Comment:
You should do this only once, not on every `Put` call. Try moving it to `NewTrie`.